source/keymanweb.js

```diff
@@ -3542,16 +3572,102 @@
+        {
+          var dirtyFlag = false; // Notes if we need to recompute our .sortedInputs array.
+
+          for(i=0; i < mutations.length; i++)
+          {
+            mutation = mutations[i];
+            
+            for(j = 0; j < mutation.addedNodes.length; j++)
+            {
+              var addedNode = mutation.addedNodes[j];
+              var lcTagName = ""
+              
+              if(addedNode.tagName) {
+                lcTagName = addedNode.tagName.toLowerCase();
+              }
+              
+              var childAdditions = [];
+
+              // Will need to handle this in case of child elements in a newly-added element with child elements.
+              if(addedNode.getElementsByTagName) 
+              {
+                var arr = addedNode.getElementsByTagName('input');
+                for(k = 0; k < arr.length; k++)
+                  childAdditions.push(arr[k]);
+
+                arr = addedNode.getElementsByTagName('textarea');
+                for(k = 0; k < arr.length; k++)
+                  childAdditions.push(arr[k]);
+
+                arr = addedNode.getElementsByTagName('iframe');
+                for(k = 0; k < arr.length; k++)
+                  childAdditions.push(arr[k]);
+              }
+
+              if(lcTagName == 'input' || lcTagName == 'textarea' || lcTagName == 'iframe')
+              {
+                dirtyFlag = true;
+                keymanweb._MutationAdditionObserved(addedNode);
+              }
+
+              for(k = 0; k < childAdditions.length; k++)
+              {
+                dirtyFlag = true;
+                keymanweb._MutationAdditionObserved(childAdditions[k]);
+              }
+            }
+
+            // There also exists a 'mutation.removedNodes' array.  We'll need to address that for issue #63.
+
+            // After all mutations have been handled, we need to recompile our .sortedInputs array.
+            if(dirtyFlag) keymanweb.listInputs();
+          }
+        });
+
+      keymanweb.observationConfig = { childList: true, subtree: true };
+      keymanweb.mutationObserver.observe(keymanweb.observationTarget, keymanweb.observationConfig);
+
     // Set exposed initialization flag to 2 to indicate deferred initialization also complete
     keymanweb['initialized']=2;
-  }  
+  }
+
+  // Used by the MutationObserver event handler to properly setup any elements dynamically added to the document post-initialization.
+  keymanweb._MutationAdditionObserved = function(Pelem)
+  {
+    if(!device.touchable)
+    {
+      // keymanweb.attachToControl is written to handle iframes, but setupDesktopElement is not.
+      if(keymanweb.setupDesktopElement(Pelem)) {
+        keymanweb.attachToControl(Pelem);
+      }
+      else if(Pelem.tagName.toLowerCase() == 'iframe') {
+        //Problem:  the iframe is loaded asynchronously, and we must wait for it to load fully before hooking in.
+        var callback = Pelem.onload;
```

Comment:
I'll look into it once we're back from the workshop, as I'll need to test the change.  iFrame's `onload` is _not_ legacy, [as it's only supported with HTML 5](http://www.dyn-web.com/tutorials/iframes/onload/), and I seem to recall choosing it because `addEventListener` wasn't sufficient for the task during development.  I need to reinvestigate that and ensure it's not an assumption.